### PR TITLE
[chore]: enable go-require rule from testifylint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

Testifylint is a linter that provides best practices with the use of testify.

This PR enables [go-require](https://github.com/Antonboom/testifylint?tab=readme-ov-file#go-require) rule from [testifylint](https://github.com/Antonboom/testifylint)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -174,3 +174,12 @@ issues:
...
     - EXC0011
+
+  # Maximum issues count per one linter.
+  # Set to 0 to disable.
+  # Default: 50
+  max-issues-per-linter: 0
+  # Maximum count of issues with the same text.
+  # Set to 0 to disable.
+  # Default: 3
+  max-same-issues: 0
```
Comment (bogdandrutu) [inline .golangci.yml:185]: why this change? No longer needed correct?

**Review by bogdandrutu — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (bogdandrutu) [reply .golangci.yml:184]: please remove ?
